### ISSUE TITLE
Validation improvements

### ISSUE THREAD

**Issue by duncanmcclean** [duncanmcclean/simple-commerce#425]
Validation seems to be something that needs to be improved. Putting rules on the blueprint doesn't work in all use cases and you may want values required on one form but not on the other. Here's my proposal to fix that:

## Proposal

My idea is that developers would be able to specify a [Form Request](https://laravel.com/docs/master/validation#creating-form-requests) when using the `{{ sc:cart:update }}` tag, maybe also for the `{{ sc:checkout }} tag, who knows.

You'd be able to specify it something like:

```
{{ sc:cart:update request="App\Http\Requests\BasicInformationRequest" }}
   <!-- whatever -->
{{ /sc:cart:update }}
```

## Implementation details

The `request` variable will likely just be another underscored form parameter, like `_redirect`.

However, for security reasons, it's probably a good idea that we start ensuring the tag's parameters aren't tampered with. 

To do this, we can create a hash when we form the outputted HTML for the tag. We can then compare the hash and the request parameters when the user submits the form to check it's all okay.

---

If anyone's got any better ideas around how to implement this, please let me know!

**Comment by jacksleight** [comment]
This sounds great.

Perhaps instead of including the full form request class name in the HTML, you instead have a list of permitted form requests in the config file, each with a name/alias. And then just include the name/alias in the HTML.

That doesn’t stop someone altering it, but it does ensure it could only be changed to a permitted request class.

**Comment by duncanmcclean** [comment]
To keep it simple, I don't think I'm going to provide a config option for it. I think the tag parameter should be enough.

**Comment by duncanmcclean** [comment]
Released in [v2.3.3](https://github.com/doublethreedigital/simple-commerce/releases/tag/v2.3.3) and [now documented](https://sc-docs.doublethree.digital/v2.3/tags#validation).
